rskj-core/src/main/java/co/rsk/peg/federation/ErpFederationArgs.java

```diff
@@ -0,0 +1,40 @@
+package co.rsk.peg.federation;
+
+import co.rsk.bitcoinj.core.BtcECKey;
+import co.rsk.bitcoinj.core.NetworkParameters;
+
+import java.time.Instant;
+import java.util.List;
+
+import static co.rsk.peg.federation.ErpFederationCreationException.Reason.NULL_OR_EMPTY_EMERGENCY_KEYS;
+
+public class ErpFederationArgs extends FederationArgs{
+    protected final List<BtcECKey> erpPubKeys;
+    protected final long activationDelay;
+    public ErpFederationArgs(
+        List<FederationMember> members,
+        Instant creationTime,
+        long creationBlockNumber,
+        NetworkParameters btcParams,
+        List<BtcECKey> erpPubKeys,
+        long activationDelay
+    ) {
+        super(members, creationTime, creationBlockNumber, btcParams);
+
+        validateEmergencyKeys(erpPubKeys);
+        this.erpPubKeys = erpPubKeys;
+        this.activationDelay = activationDelay;
+    }
+
+    public static ErpFederationArgs fromFederationArgs(FederationArgs federationArgs, List<BtcECKey> erpPubKeys, long activationDelay){
```

Comment:
Any reason to make this static instead of a contructor that receives FederationArgs?

Reply:
I proposed it this way to abstract how FederationArgs objects are built. It may not be seen as a big deal at first glance, but some sources state the `new` operator may be harmful in the long term because of the cost of changing a given implementation.

Source: https://www.baeldung.com/creational-design-patterns